x/chainlink/keeper/keeper.go

```diff
@@ -34,3 +38,95 @@ func NewKeeper(
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+func (k Keeper) SetFeedData(ctx sdk.Context, feedData *types.MsgFeedData) {
+	// TODO: add more complex feed validation here such as verify against other modules
+
+	// TODO: deserialize the feedData.FeedData if it's an OCR report, assume all the feedData is OCR report for now.
+	// this is simulating the OCR report deserialization lib
+	/****************/
+	observations := make([]*types.Observation, 0, len(feedData.GetFeedData()))
+	for _, b := range feedData.GetFeedData() {
+		o := &types.Observation{Data: []byte(string(b))}
+		observations = append(observations, o)
+	}
+	deserializedOCRReport := types.OCRAbiEncoded{
+		Context:      []byte("testcontext"),
+		Oracles:      feedData.Submitter.Bytes(),
+		Observations: observations,
+		Submitter:    feedData.GetSubmitter(),
+	}
+	/****************/
+	// TODO: verify deserializedOCRReport here
+
+	finalFeedDataInStore := types.OCRFeedDataInStore{
+		FeedData:              feedData,
+		DeserializedOCRReport: &deserializedOCRReport,
+		RoundId:               roundId,
+	}
+	// TODO: add round index here, need to figure out a way to generate round number, below line is a tmp solution
+	roundId += 1
+
+	// use store with gas meter
+	store := ctx.KVStore(k.storeKey)
+
+	f := k.cdc.MustMarshalBinaryBare(&finalFeedDataInStore)
+	store.Set(types.KeyPrefix(types.FeedDataKey), f)
```

Comment:
add return 
```
return ctx.BlockHeight(), ctx.TxBytes()
```

Reply:
```suggestion
	store.Set(types.KeyPrefix(types.FeedDataKey), f)
	return ctx.BlockHeight(), ctx.TxBytes()
```

Reply:
are you sure `ctx.TxBytes()` is the txHash?

Reply:
not exactly sure, we can test it out but i would think that it would be because we're still using the same value anyway in your implementation
```
string(ctx.TxBytes())
```